Makefile

```diff
@@ -388,7 +393,7 @@ validate-gen-versions:
 	make dirty-check
 
 ## Deploys images to registry
-cd:
+cd: image-all
```

Comment:
If anyone ever has to do cd manually it might be useful to have the `cd` target not invoking the `image-all` target. WDYT?

Reply:
cd doesn't make sense without image-all though, it really is a requirement of calling cd.

Reply:
That means though that running cd will always build the images, it won't be possible to separate those steps. Ideally there would be a file indicator or way to check that the images exist rather than having to invoke `image-all` on any invocation of `cd`.
But we can always change this in the future, if we split building and pushing of images into separate steps.

Reply:
Running CD should always build the images if they're not there, it is a dependency of CD. If image-all has already been called then it should be quick (rebuilding docker images when nothing has changed is pretty immediate). It could be skipped entirely using some other method (possibly the file descriptor method but I'm not a huge fan of that), but again, rebuilding a docker image where nothing has changed is extremely fast (docker got smart about this).